### PR TITLE
NO-ISSUE: E2E Full harness (with API) + test for checking TPM status reporting

### PR DESCRIPTION
**Pull request by mangelajo**
This PR extends the basic E2E harness including the admin API
client, and moves the two e2e agent tests to the new harness

As an example where the API is required introduces a new test that:

* Verifies the TPM reporting from within the VM

This also switches the runner for E2E to the ginkgo binary which
produces better reporting console output as well as better junit output.

### CODE REVIEW

File: test/harness/e2e/vm/domain-template.xml
Comment (mangelajo) [inline test/harness/e2e/vm/domain-template.xml:21]: 😎

File: test/harness/e2e/vm/domain-template.xml
Comment (mangelajo) [inline test/harness/e2e/vm/domain-template.xml:22]: I wonder if this could be "yes" ... checking

Comment (mangelajo) [reply test/harness/e2e/vm/domain-template.xml:22]: oh it works, nice :)

File: test/harness/e2e/vm/domain-template.xml
Comment (mangelajo) [reply test/harness/e2e/vm/domain-template.xml:21]: Hmmm not sure if this is making it fail to boot on github CI :-( checking

Comment (mangelajo) [reply test/harness/e2e/vm/domain-template.xml:26]: 4 days later... https://github.com/flightctl/flightctl/pull/231/files#diff-29fa4e9e9dfff8f099fe2fdc77e1e702165851b9a835991d8d4a1f7cd14a71beR17

File: test/harness/e2e/vm/domain-template.xml
Comment (mangelajo) [inline test/harness/e2e/vm/domain-template.xml:37]: fixing typo, this shouldn't be specific about the pts... libvirt (or qemu) will do auto-allocation

File: test/test.mk
Comment (mangelajo) [inline test/test.mk:28]: I switched to ginkgo runner because it provides better output of what's going on, and better junit output too, we may consider also switching the integration test output for the same reason.

File: test/harness/e2e/vm/vm_linux.go
Comment (mangelajo) [inline test/harness/e2e/vm/vm_linux.go:309]: Not explicitly required for this patch. But when I expect that some tests will require running the VM without waiting to SSH.

File: test/harness/e2e/vm/vm_linux.go
Comment (mangelajo) [inline test/harness/e2e/vm/vm_linux.go:129]: We need to clear the buffer after reading.

**Comment by mangelajo** [comment]
> https://github.com/flightctl/flightctl/actions/runs/8708012675/job/23884384875?pr=231#step:9:4336

Thanks, yes... 
https://github.com/flightctl/flightctl/pull/231/files#diff-08d1d54287572446320e640945c840e8b94aebdad92c871d5e344270fe2a0d4dR128
I had to clear a buffer after reading ... otherwise depending on timing, the tests waiting for specific console output would get confused

File: test/e2e/agent_test.go
Comment (hexfusion) [inline test/e2e/agent_test.go:16]: just curious why we need to double the timeout?

File: test/harness/e2e/harness.go
Comment (hexfusion) [inline test/harness/e2e/harness.go:41]: can you return an empty string and then validate the return to maybe not sure if this would just work
```
Fail("Could not find top-level directory")
```

File: test/e2e/agent_test.go
Comment (hexfusion) [inline test/e2e/agent_test.go:110]: should we log the original error?

File: test/e2e/agent_test.go
Comment (hexfusion) [inline test/e2e/agent_test.go:105]: can we add a comment on what is happening here why are we testing a mutated copy of PCR?

**Review by hexfusion — COMMENTED** [review]
looks good few notes

File: test/e2e/agent_test.go
Comment (mangelajo) [reply test/e2e/agent_test.go:16]: The TPM test is more complex and takes longer. (Waiting for enrollment, approval, waiting for device report to check again the TPM details)... 

We could bump that case , I am trying to remember and may be some of the other tests also timed out. We could be more specific per test I guess.

File: test/e2e/agent_test.go
Comment (mangelajo) [reply test/e2e/agent_test.go:105]: Ack, I will clarify what's going on here. Thanks for bringing it up.

File: test/e2e/agent_test.go
Comment (mangelajo) [reply test/e2e/agent_test.go:110]: Hmm, you are right; an error at this point is probably worth logging, at least. At least with the current usage, it should not happen, I believe.

File: test/harness/e2e/harness.go
Comment (mangelajo) [reply test/harness/e2e/harness.go:41]: Thanks. I remember writing that and thinking... I am sure there is a better way :D

File: test/e2e/agent_test.go
Comment (mangelajo) [reply test/e2e/agent_test.go:110]: Ok, I checked, and I decided to go a little bit harder. The test will fail on errors here. As that should not be happening.

**Review by hexfusion — APPROVED** [review]

**Pull request merged**
